### ISSUE TITLE
Provider v3.64.x and aws_network_interface data source error

### ISSUE THREAD

**Issue by kabalman** [hashicorp/terraform-provider-aws#21693]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform CLI version - 0.14 to 1.0.4 (not tested on higher versions than 1.0.4)
Provider aws ve4rsion 3.64.x (3.64.0, 3.64.1 and 3.64.2)

### Affected Resource(s)

aws_network_interface Data source (and possibly resource as well)

### Terraform Configuration Files

```hcl

data "aws_network_interface" "eni1" {
  id = "eni-a1b2c3d4"
}

```

### Debug Output

terraform plan produces the following error:

```
Error: error setting attachment: Invalid address to set: []string{"attachment", "0", "instance"}

  on main.tf line 25, in data "aws_network_interface" "eni1":
  25: data "aws_network_interface" "eni1" {


Releasing state lock. This may take a few moments... 
```

### Panic Output

None

### Expected Behavior

Terraform should have read the ENI listed in the aws_network_interface data source, but it errors. This was working until AWS provider version v3.63.0

### Actual Behavior

terraform plan produces the following error:
```
Error: error setting attachment: Invalid address to set: []string{"attachment", "0", "instance"}

  on main.tf line 25, in data "aws_network_interface" "eni1":
  25: data "aws_network_interface" "eni1" {


Releasing state lock. This may take a few moments...
```

BTW - I see the same error in the DEBUG logs for provider version 3.63.0, but it does not fail the plan though. The data source is read successfully and the code works fine.
```
[ERROR] setting state: Invalid address to set: []string{"attachment", "0", "instance"}: timestamp=2021-11-09T17:29:25.890Z
```

### Steps to Reproduce

Code:
```hcl
data "aws_network_interface" "eni1" {
  id = "eni-a1b2c3d4"
}
```

1. `terraform plan`

### Important Factoids

None

### References

None

**Comment by ewbankkit** [comment]
@kabalman Thanks for raising this issue.
It has already been noticed in #21647. I'm going to close this one as a duplicate so that we can concentrate discussion in the linked issue.
Please add any additional comments there.

**Comment by ewbankkit** [comment]
This is fixed in **v3.65.0** of the Terraform AWS Provider, scheduled to be released this Thursday, November 11th.
